File: javascript/VEdge.js

function VEdge(s, a, b)		// start, left, right
{
	this.left = a;		// point on left
	this.right = b;		// point on right
	
	this.start = s;		// start point
	this.end = null;	// end point
	
	this.f = (b.x - a.x) / (a.y - b.y);
	this.g = s.y - this.f*s.x;
	this.direction = new Point(b.y-a.y, -(b.x - a.x));
	this.B = new Point(s.x+this.direction.x, s.y + this.direction.y);	// second point of line
	
	this.intersected = false;
	this.iCounted = false;
	
	this.neighbour = null;
}

File: ported/VEdge.py
__author__ = 'sean'
from Point import Point

class VEdge:
    def __init__(self, s, a, b):
        self.left = a
        self.right = b
        self.start = s
        self.end = None

        self.f = (b.x - a.x) / (a.y - b.y)
        self.g = s.y - self.f * s.x
        self.direction = Point(b.y-a.y, -(b.x-a.x))
        self.B = Point(s.x + self.direction.x, s.y + self.direction.y)

        self.interesected = False
        self.iCounted = False
        self.neighbour = None